blog.md

```diff
@@ -1,26 +1,78 @@
+
+I also span an additional three [CoreOS](https://coreos.com) nodes using [Terraform](https://terraform.io/) on which I ran [Jenkins](http://jenkins-ci.org/) using the standard [jenkins docker image](https://hub.docker.com/_/jenkins/).
+
+To demonstrate a backing service for my application I also ran a Percona database cluster (one on each of the CoreOS nodes) using the [paulczar/percona-galera](https://github.com/paulczar/docker-percona_galera) docker image which uses [etcd](https://coreos.com/etcd/) to find each other and form automatically create the galera replication set. I also deployed a mysql proxy using the [paulczar/maxscale](https://hub.docker.com/r/paulczar/maxscale/) image which discovers the databases via [etcd](https://coreos.com/etcd/).
+
+## Ghost
+
+The application I chose to demo is the [Ghost blogging platform](https://ghost.org/download/) (literally the blog you are reading right now). I chose it because it's a fairly simple app with well-known backing service (MySQL). The source including my `Dockerfile` and customizations can be found in the [paulczar/ci-demo](https://github.com/paulczar/ci-demo) github repository.
+
+The database backend for ghost is hosted on the percona cluster I have running on the CoreOS nodes and the location and credentials for it are passed into Ghost via environment variables set by the Deis cli when creating the applications.
+
+I wanted to stick to the [GitHub Flow](https://guides.github.com/introduction/flow/) for development as much as possible (the merge/deploy steps are a bit different here, but should be easy enough to follow along with) utilizing GitHub's notification system on Pull Requests and Merges to trigger Jenkins jobs.
+
+I created a `development` branch in the git repository for the application and then used the Deis cli to create two applications [ghost](http://ghost.ci-demo.paulcz.net) from the code in the `master` branch and [stage-ghost](http://stage-ghost.ci-demo.paulcz.net) from the development branch to be my Production and Staging environments.
+
+Both the `development` and `master` branches are protected with GitHub settings that restrict changes from being pushed directly to the branch and any Pull Requests need to pass tests before they can be merged.
+
+## Deis
+
+Deploying applications with Deis is quite easy and very similar to deploying applications to Heroku. As long as your git repo has a `Dockerfile` (or supports being discovered by the [cedar](https://devcenter.heroku.com/articles/cedar) tooling) Deis will figure out what needs to be done to run your application.
 
-## Application
+After creating an application with the Deis cli you simply push to a remote git server managed by Deis to handle uploading code to install/upgrade the application:
```

Comment:
This sentence is difficult to parse. To be honest, I'm not sure exactly what you're saying here, so I don't have an immediate suggestion.